wrapanapi/ec2.py

```diff
@@ -702,3 +702,114 @@ def delete_topic(self, arn):
         except Exception:
             self.logger.exception("Delete of {} topic failed.".format(arn))
             return False
+
+    def volume_exists_and_available(self, volume_name=None, volume_id=None):
+        """
+        Method for checking existence and availability state for volume
+
+        Args:
+            volume_name: Name of volume, if not set volume_id must be set
+            volume_id: ID of volume in format vol-random_chars, if not set volume_name must be set
+
+        Returns:
+            True if volume exists and is available.
+            False if volume doesn't exist or is not available.
+        """
+        if volume_id:
+            try:
+                response = self.ec2_connection.describe_volumes(
+                    VolumeIds=[volume_id],
+                    Filters=[
+                        {
+                            'Name': 'status',
+                            'Values': ['available']
+                        }
+                    ]
+                )
+                if response.get('Volumes'):
+                    return True
+                else:
+                    return False
+            except:
+                return False
+        elif volume_name:
+            response = self.ec2_connection.describe_volumes(
+                Filters=[
+                    {
+                        'Name': 'status',
+                        'Values': ['available']
+                    },
+                    {
+                        'Name': 'tag:Name',
+                        'Values': [volume_name]
+                    }
+                ]
+            )
+            if response.get('Volumes'):
+                return True
+            else:
+                return False
+        else:
+            raise Exception("Neither volume_name nor volume_id were specified.")
```

Comment:
This exception type should be TypeError, the same like when you invoke a function with incorrect number of arguments.